packages/ws-server/src/webSocketServer.ts

```diff
@@ -175,71 +176,81 @@ app.ws.use(async (ctx) => {
     methodsCounter.labels(method).inc();
     methodsCounterByIp.labels(ctx.request.ip, method).inc();
 
-    if (method === constants.METHODS.ETH_SUBSCRIBE) {
-      if (limiter.validateSubscriptionLimit(ctx)) {
-        const event = params[0];
-        const filters = params[1];
-        let subscriptionId;
-
-        if (event === constants.SUBSCRIBE_EVENTS.LOGS) {
-          try {
-            await validateSubscribeEthLogsParams(filters, requestIdPrefix);
-          } catch (error) {
-            logger.error(
-              error,
-              `${connectionIdPrefix} ${requestIdPrefix} Encountered error on ${
-                ctx.websocket.id
-              }, method: ${method}, params: ${JSON.stringify(params)}`,
-            );
-            response = jsonResp(request.id, error, undefined);
-            ctx.websocket.send(JSON.stringify(response));
-            return;
-          }
+    // Early return if subscription limit is exceeded
+    if (method === constants.METHODS.ETH_SUBSCRIBE && !limiter.validateSubscriptionLimit(ctx)) {
```

Comment:
need a test for this

Reply:
There was an existing test to cover the subscription limit. I just added one for subscription limit and `newHeads`.